substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/linux/LinuxSubstitutions.java

```diff
@@ -66,6 +72,48 @@ public static String mapLibraryName(String libname) {
     }
 }
 
+@Platforms(Platform.LINUX.class)
+@TargetClass(java.lang.Runtime.class)
+final class Target_java_lang_Runtime {
+
+    @Substitute
+    public int availableProcessors() {
+        if (SubstrateOptions.MultiThreaded.getValue()) {
+            int res = (int) Unistd.sysconf(Unistd._SC_NPROCESSORS_CONF());
```

Comment:
This is not the right CPU count to check. You need to obtain the number of *possible* CPUs, which is the upper limit the system can support after CPU hotplug. The easiest way to do this is to call `sched_getaffinity` repeatedly with growing buffer sizes, unfortunately.

Reply:
Fair enough.  Fix forthcoming...

Reply:
Updated.

Reply:
By the way, I think you should be able to test this with `possible_cpus=5000` or similar on the kernel command line, and a Red Hat Enterprise Linux or CentOS kernel (Debian won't work because the hard limit is 512 CPUs). No need to find a system with a broken firmware that says it can be hot-plugged to these many CPUs.